### PR TITLE
Add host argument to `dev` and `test` commands

### LINKED ISSUE #193: Add `--host` option to `carton dev` and `carton test`
Job to be done:
Host Carton Dev inside a container when I am testing out solutions.

Current behavior:
Carton hosts on 127.0.0.1 which doesn't export when using docker

Expected behavior:
have the host as a variable (just like the port) so I can choose where to host.

```
carton dev -host 0.0.0.0 -p 9000
````

### PR DESCRIPTION
**Pull request by literalpie**
Closes #193

### CODE REVIEW

File: Sources/CartonCLI/Commands/Test.swift [hunk 1]
```diff
@@ -60,6 +60,12 @@ struct Test: ParsableCommand {
   )
   var port = 8080
 
+  @Option(
+    name: .shortAndLong,
+    help: "Set the location where the testing server will run."
```
Comment (MaxDesiatov) [inline Sources/CartonCLI/Commands/Test.swift:65]:
```suggestion
    help: "Set the location where the testing server will run. Default is `127.0.0.1`"
```

File: Sources/CartonCLI/Commands/Dev.swift [hunk 2]
```diff
@@ -47,6 +47,9 @@ struct Dev: ParsableCommand {
   @Option(name: .shortAndLong, help: "Set the HTTP port the development server will run on.")
   var port = 8080
 
+  @Option(name: .shortAndLong, help: "Set the location where the development server will run.")
```
Comment (MaxDesiatov) [inline Sources/CartonCLI/Commands/Dev.swift:50]:
```suggestion
  @Option(name: .shortAndLong, help: "Set the location where the development server will run. Default is `127.0.0.1`.")
```

**Review by MaxDesiatov — COMMENTED** [review]
Overall this is great, thanks for the contribution! Just a minor suggestion for highlighting the default in the option help output.

**Comment by MaxDesiatov** [comment]
@thecb4 would you have a moment to review changes to your testing code?

**Comment by MaxDesiatov** [comment]
I'm merging this as is, as it fixes CI build failures. @thecb4 feel free to add any comments to the diff, which we can address in a seperate PR.

**Review by MaxDesiatov — APPROVED** [review]

**Review by thecb4 — COMMENTED** [review]
Sounds good. I will add back the cleanup.

File: Tests/CartonCommandTests/DevCommandTests.swift [hunk 3]
```diff
@@ -113,8 +150,5 @@ final class DevCommandTests: XCTestCase {
       print("no response from server after \(count) tries or \(Int(count) * Int(delay)) seconds")
       XCTFail("Could not reach server")
     }
-
-    // clean up
-    do { try packageDirectory.appending(component: ".build").delete() } catch {}
```
[the remark this answers is not in the document]
Comment (MaxDesiatov) [reply Tests/CartonCommandTests/DevCommandTests.swift:118]: Sure, a separate PR for that is perfectly fine 🙂

[the remark this answers is not in the document]
Comment (literalpie) [reply Tests/CartonCommandTests/DevCommandTests.swift:118]: Oh no! So sorry I missed that

**Pull request merged**